cmake/LibMPI.cmake

```diff
@@ -0,0 +1,217 @@
+# This file contains CMake code related to MPI. (Taken from the
+# ParallelIO project.)
+
+# Jim Edwards
+include (CMakeParseArguments)
+
+# Find Valgrind to perform memory leak check
+if (PIO_VALGRIND_CHECK)
+  find_program (VALGRIND_COMMAND NAMES valgrind)
+  if (VALGRIND_COMMAND)
+    set (VALGRIND_COMMAND_OPTIONS --leak-check=full --show-reachable=yes)
+  else ()
+    message (WARNING "Valgrind not found: memory leak check could not be performed")
+    set (VALGRIND_COMMAND "")
+  endif ()
+endif ()
+
+#
+# - Functions for parallel testing with CTest
+#
+
+#==============================================================================
+# - Get the machine platform-specific
+#
+# Syntax:  platform_name (RETURN_VARIABLE)
+#
+function (platform_name RETURN_VARIABLE)
+
+  # Determine platform name from site name...
+  site_name (SITENAME)
+  
+  # hera
+  if (SITENAME MATCHES "^hfe01" OR
+      SITENAME MATCHES "^hfe02" OR
+      SITENAME MATCHES "^hfe03" OR
+      SITENAME MATCHES "^hfe04" OR
+      SITENAME MATCHES "^hfe05" OR
+      SITENAME MATCHES "^hfe06" OR
+      SITENAME MATCHES "^hfe07" OR
+      SITENAME MATCHES "^hfe08" OR
+      SITENAME MATCHES "^hfe09" OR
+      SITENAME MATCHES "^hfe10" OR
+      SITENAME MATCHES "^hfe11" OR
+      SITENAME MATCHES "^hfe12")
```

Comment:
How managable is this practice to extend to the various machines?
This is a hack!

Reply:
It came from PIO's MPI tests and then I adapted it and took code from the weather model's detect machine script and put it into CMake. I agree, probably not the best solution.

Reply:
We should not "script" in cmake.
I see a lot of practice of writing scripts in cmake.
One can boil this down to a function and a few command line arguments as input and not needing any of these added files.

Reply:
For extensibility: the code looks for certain machine names and then does a default answer for the rest. 

For CMake improvements: I agree. I cribbed this cmake code from the PIO project, and it works, which is nice. But I'm happy to take improvements. However, I am not enough of a cmake expert to attempt this at this time...